### ISSUE TITLE
space-before-keywords in jsx context (bug)

### ISSUE THREAD

**Issue by remitbri** [eslint/eslint#3789]
In js, 
```js
/*eslint space-before-keywords: [2, "always"]*/
(function() {})()
```
is not considered an error.

But in jsx
```jsx
/*eslint space-before-keywords: [2, "always"]*/
onClick={function foo() {…}}
```
flags one (using v1.4.1)

**Comment by ilyavolodin** [comment]
Could you provide console output of the error please?

**Comment by remitbri** [comment]
`Missing space before keyword "function". (space-before-keywords)`

**Comment by mmrko** [comment]
hmm, maybe `space-before-keywords` should not concern itself with spaces between `{` and keywords and instead let `block-spacing` handle them...

**Comment by nzakas** [comment]
Sounds good. You should double-check how this works right now outside of JSX.

**Comment by mmrko** [comment]
Ok, great! Will do.
